packages/loot-core/src/types/models/transaction.d.ts

```diff
@@ -26,7 +26,11 @@ export interface NewTransactionEntity {
   subtransactions?: Omit<NewTransactionEntity, 'account' | 'date'>[];
 }
 
-export interface TransactionEntity extends NewTransactionEntity {
+export interface TransactionEntity
+  extends Omit<
+    NewTransactionEntity,
+    'account' | 'category' | 'payee' | 'schedule' | 'subtransactions'
+  > {
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏aren't these types overridden? 🤔

Reply:
Apparently  you can't incompatably override. Rather than omiting we might be able to mark them here as taking a string OR their entity, or we could add a new base type that allows either, then build the other two types off of it. Example of the error we get:

```
node_modules/@actual-app/api/@types/loot-core/types/models/transaction.d.ts:29:18 - error TS2430: Interface 'TransactionEntity' incorrectly extends interface 'NewTransactionEntity'.
  Types of property 'account' are incompatible.
    Type 'AccountEntity' is not assignable to type 'string'.
      Type '{ id: string; name: string; offbudget: 0 | 1; closed: 0 | 1; sort_order: number; tombstone: 0 | 1; } & _SyncFields<true>' is not assignable to type 'string'.
```

This is another example of an issue with our current setup where our .d.ts files are avoiding being validated by tsc within the project. I assumed the fix was just using .ts files instead of .d.ts and letting tsc generate them. But I haven't looked into whether we can just have it run validation on our .d.ts  files instead.

Reply:
Thanks for the explanation!